### PR TITLE
platform dependent @Formula and @Where annotations

### PR DESCRIPTION
**Pull request by rPraml**
So it's November and Java 8 time :+1: (yes, this featue needs java 8 on a few places)

So... I tried to improve the repeatable annotation handling:
- AnnotationBase.get() was extended to resolve repeatable (and pseudo-repeatable) annotations automatically. This makes it obsolete to manually check the container annotation(JoinColumns, Indices, NamedQueries,....)
- there is also a .getAll() and a .getPlatformMatchingAnnotation() method now.
- @Formula and @Where can be specified now per platform. (e.g. different formula for Pstgres & Mysql)
- When multiple size-annotations are present, the max value is determined over all @Size annotations
- Enhancement in @NotNull annotation - currently there is no possibility to exclude them from validation, but include the @notNull information is in the DDL (I need this for some auto-generated properties, that are set after successful validation)

cheers
Roland

### CODE REVIEW

File: src/main/java/com/avaje/ebeaninternal/server/deploy/parse/AnnotationAssocOnes.java [hunk 1]
```diff
@@ -113,14 +112,18 @@ private void readAssocOne(DeployBeanPropertyAssocOne<?> prop) {
...
-    if (joinColumns != null) {
-      prop.getTableJoin().addJoinColumn(false, joinColumns.value(), beanTable);
-    }
 
     JoinTable joinTable = get(prop, JoinTable.class);
     if (joinTable != null) {
-      prop.getTableJoin().addJoinColumn(false, joinTable.joinColumns(), beanTable);
+      for (JoinColumn joinColumn : joinTable.joinColumns()) {
+        prop.getTableJoin().addJoinColumn(false, joinColumn, beanTable);
+        if (!joinColumn.updatable()) {
```
Comment (rPraml) [inline src/main/java/com/avaje/ebeaninternal/server/deploy/parse/AnnotationAssocOnes.java]: I'm not sure if this is correct.

**Comment by rPraml** [comment]
Can you take a look at this failing build in "TestHistoryInsert" https://travis-ci.org/ebean-orm/ebean/jobs/173188975
I cannot explain why it fails. Maybe a timing problem?

**Comment by 0xbaadf00d** [comment]
Is the issue fixed? Maybe try to close issue and reopen it to force TravisCI to restart job

File: src/main/java/com/avaje/ebeaninternal/server/deploy/parse/AnnotationParser.java [hunk 2]
```diff
@@ -52,26 +54,15 @@ protected void setCascadeTypes(CascadeType[] cascadeTypes, BeanCascadeInfo casca
    */
   protected void readEmbeddedAttributeOverrides(DeployBeanPropertyAssocOne<?> prop) {
 
-    AttributeOverrides attrOverrides = get(prop, AttributeOverrides.class);
-    if (attrOverrides != null) {
+    Set<AttributeOverride> attrOverrides = getAll(prop, AttributeOverride.class);
```
Comment (rPraml) [inline src/main/java/com/avaje/ebeaninternal/server/deploy/parse/AnnotationParser.java]: The new "getAll" method should also have fixed #878

**Pull request merged**